OAuthSwift/String+OAuthSwift.swift

```diff
@@ -32,10 +32,7 @@ extension String {
     }
 
     func urlEncodedStringWithEncoding(encoding: NSStringEncoding) -> String {
-        let originalString: NSString = self
-        let customAllowedSet =  NSCharacterSet(charactersInString:" :/?&=;+!@#$()',*=\"#%/<>?@\\^`{|}[]").invertedSet
-        let escapedString = originalString.stringByAddingPercentEncodingWithAllowedCharacters(customAllowedSet)
-        return escapedString! as String
+        return CFURLCreateStringByAddingPercentEscapes(kCFAllocatorDefault, self, nil, "!*'\"();:@&=+$,/?%#[] ", CFStringBuiltInEncodings.UTF8.rawValue) as String
```

Comment:
Turns out the original implementation did not use the `encoding` either - and always passes in `OAuthSwiftDataEncoding`, which is `NSUTF8StringEncoding` anyway. So I am not going to change this implementation.